WordPress/src/main/java/org/wordpress/android/ui/plans/PlanUpdateService.java

```diff
@@ -0,0 +1,181 @@
+package org.wordpress.android.ui.plans;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+
+import com.android.volley.VolleyError;
+import com.wordpress.rest.RestRequest;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.ui.plans.models.SitePlan;
+import org.wordpress.android.ui.prefs.AppPrefs;
+import org.wordpress.android.util.AppLog;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.greenrobot.event.EventBus;
+
+/**
+ * service which updates both global plans and available plans for a specific site
+ */
+
+public class PlanUpdateService extends Service {
+
+    private static final String ARG_LOCAL_BLOG_ID = "local_blog_id";
+    private int mNumActiveRequests;
+    private final List<SitePlan> mSitePlans = new ArrayList<>();
+
+    public static void startService(Context context, int localTableBlogId) {
+        Intent intent = new Intent(context, PlanUpdateService.class);
+        intent.putExtra(ARG_LOCAL_BLOG_ID, localTableBlogId);
+        context.startService(intent);
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        AppLog.i(AppLog.T.PLANS, "plan update service > created");
+    }
+
+    @Override
+    public void onDestroy() {
+        AppLog.i(AppLog.T.PLANS, "plan update service > destroyed");
+        super.onDestroy();
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        int localBlogId = intent.getIntExtra(ARG_LOCAL_BLOG_ID, 0);
+
+        mNumActiveRequests = 3;
```

Comment:
Plans Service is not stopped when the user exits the Plans activity, this could lead to unespected results on slow network connections. 
Step to repro:
- Tap on `Plans`
- Tap back, switch site.
- Tap on `Plans` again
A new call to ` PlanUpdateService.startService(this, mLocalBlogID);` with a new blogId is made, but previous network connections are still alive. Resulting in reporting back to the Plan activity wrong data.
![screen shot 2016-02-26 at 15 34 22](https://cloud.githubusercontent.com/assets/518232/13355092/e869ab86-dc9e-11e5-935c-90a3d7b8b17d.png)